### DISCUSSION TITLE
Custom auth plugin

### DISCUSSION THREAD

**Discussion by jb-san** [verdaccio/verdaccio#3211] [Q&A] [answered]
Hi, I wondered what the suggested way to develop a custom auth plugin and test it locally is.
is there a way to develop and bundle the plugin in a docker image that can then be deployed?
I did find https://github.com/verdaccio/verdaccio/issues/1677
but it is light on information, the example https://github.com/barolab/verdaccio-minio/blob/master/example/Dockerfile#L10
still needs you to install a plugin globally and then `ln -s` it into the container
right now, the only way I have found is to create a custom docker file and copy the built plugin and configuration during the docker image build. is there a better way?

I think documenting this could be useful for 
1. when developing a plugin that you don't plan to publish to the official npm registry
2. during local development of plugins

thanks in advance

**Comment by juanpicado** [comment]
I will write an website page about it in the next days/weeks that resume all possible ways.

**Comment by juanpicado** [comment, marked as answer]
Check https://github.com/verdaccio/verdaccio/pull/3217 that will be useful for you.

Here https://github.com/verdaccio/verdaccio/pull/3217/files#diff-eeceea1c0ec2bdf954e72212f5d575ba7f0f3f77666fb69fd1b2e3c80eedf0a3
